docs/advanced/errors.md

```diff
@@ -27,6 +27,86 @@ export class MyErrorHandler implements ErrorHandler {
 export class AppModule {}
 ```
 
+## Handling errors within an `@Select`
+
+```ts
+@Component({
+  /* .. */
+})
+class AppComponent {
+  @Select(state => state.count.number.value)
+  public count$: Observable<number>;
+}
+```
+
+One of the main problems is that if your application state is not correct later.
+
+```ts
+this.store.reset({}); // destroy all states
+```
+
+Then automatically all subscribers will be unsubscribed from the store, as an internal error will occur.
+To track such things, you must disable `suppressErrors` flag:
+
+```ts
+@NgModule({
+  imports: [
+    NgxsModule.forRoot([CountState], {
+      selectorOptions: {
+        suppressErrors: false // default by true
+      }
+    })
+  ]
+})
+export class AppModule {}
+```
+
+This way you can track application errors and add error handling.
+
+```ts
+@Component({
+  /* .. */
+})
+class AppComponent {
+  @Select(state => {
+    try {
+      return state.count.number.value;
+    } catch (err) {
+      console.log('error', err);
+      // throw err;
+      // If you forward an error further, then an automatic unsubscription from the stream will occur.
+      // Do not do this if you do not want it.
+    }
+  })
+  public count$: Observable<number>;
+}
+```
+
+#### Why does Rxjs unsubscribe on error?
+
+An explanation of why this matters can be found from Rx design guidelines:
+
+The single message indicating that an observable sequence has finished ensures that consumers of the observable sequence can deterministically establish that it is safe to perform cleanup operations.
+
+A single failure further ensures that abort semantics can be maintained for operators that work on multiple observable sequences.
+
+In short, if you want your observers to keep listening to the store after a error has occurred, do not deliver that error to the store, but rather handle it in some other way (e.g. use catch, retry or deliver the error to a dedicated subject).
+
+```ts
+this.store.select(CountState).pipe(
+  retryWhen(errors =>
+    errors.pipe(
+      // log error message
+      tap(val => console.log(`Value ${val} was too high!`)),
+      // restart in 6 seconds
+      delayWhen(val => timer(val * 1000))
```

Comment:
Maybe some example without `delayWhen`? I think it's redundant. You could provide some example from the real app. So the reader will not have questions "what is `delayWhen` doing here?"

Reply:
done